package/pyproject.toml

```diff
@@ -0,0 +1,12 @@
+[build-system]
+# Minimum requirements for the build system to execute
+requires = [
+    "Cython>=0.28",
+    # lowest NumPy we can use for a given Python,
+    # except for more exotic platforms (linux/Mac ARM flavors)
+    "numpy==1.18; python_version=='3.7' and (platform_machine!='arm64' or platform_system!='Darwin') and platform_machine!='aarch64'",
```

Comment:
Also I'm not sure I fully understand your comments about going to a lower numpy version. Given that we're meant to be sticking to NEP29 going forward, probably all we'll want to do is bump things up as necessary?

Reply:
You always want to build with the lowest possible NumPy version because there is forward but not backward ABI compatibility at runtime.

That's also why we should probably just use: https://github.com/scipy/oldest-supported-numpy (that leaves the decisions about minimum viable NumPy up to the experts) See also, i.e., pandas example:
https://github.com/pandas-dev/pandas/blob/main/pyproject.toml#L8

What's the timeline for getting this stuff ready?

Reply:
Ideally I'd like to cut a release soon-ish (week or two max). If it doesn't make it then it can always go in 2.2.0. (we're going to try to trim releases so they are closer to every 3 months).

I'm not sure I understand how this works though. If someone calls `python setup.py install` without any dependencies then will it pull the lowest numpy version and stick to that only?

Reply:
`setup.py install` is effectively deprecated (though should still work for quite some time), and shouldn't be affected by `pyproject.toml`

For building MDAnalysis with a modern approach (i.e., `pip install .`)--`pip` will create an isolated build environment with the stuff we specify in this file, and build MDAnalysis in a manner that should be maximally compatible with whatever runtime environment the user has--whether that be new or old NumPy. An obvious application is building a binary wheel that won't barf with older versions of NumPy, within reason.

Reply:
Sorry took me so much time to get back to this.

> pip will create an isolated build environment

🤯 Thanks for explaining this @tylerjereddy, I had no idea that was a thing.

In terms of contents of this PR I think it all makes sense now.